### PR TITLE
PaperUI: enable trigger channel linking

### PR DESCRIPTION
**Pull request by htreu**
This enables trigger channel linking through PaperUI.

* Trigger channels can only link to existing items
* On-the-fly item creation is not possible for Trigger channels
* Also _dummy_  items will not be created by ChannelItemProvider
* In _simple mode_ the same item selection pops up as in advanced mode

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java [hunk 1]
```diff
@@ -265,6 +301,29 @@ private void createItemForLink(ItemChannelLink link) {
         }
     }
 
+    private Collection<String> getItemType(Channel channel) {
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java]: Can we change the method name to something that state what the implementation is doing (and perhaps we should use plural)?
For example `getProfileItemTypes` or something similar.
Sure, it is a private method, but it should be clear that the item types depend on the profiles and not on the accepted item type of the channel (without reading the implementation).

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java [hunk 2]
```diff
@@ -41,4 +42,20 @@
...
+     *
+     * Please note:
+     * <ul>
+     * <li>This will override any default behavior
+     * <li>A "profile" configuration on the link will override this suggestion
+     * </ul>
+     *
+     * @param channelType the {@link ChannelType} of the linked channel
+     * @param itemType the linked itemType (not applicable for trigger channels)
+     * @return the profile identifier or {@code null} if this advisor
```
Comment (triller-telekom) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java]: "if this advisor" ...? something is missing here

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java [hunk 3]
```diff
@@ -41,4 +42,20 @@
...
+     * Suggest a custom profile for a given {@link ChannelType} (and potentially also the itemType).
+     *
+     * Please note:
+     * <ul>
+     * <li>This will override any default behavior
+     * <li>A "profile" configuration on the link will override this suggestion
+     * </ul>
+     *
+     * @param channelType the {@link ChannelType} of the linked channel
+     * @param itemType the linked itemType (not applicable for trigger channels)
```
Comment (triller-telekom) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java]: I thought trigger channels can now be linked to items, can't they?

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers/controller.things.spec.js [hunk 4]
```diff
@@ -175,51 +203,51 @@ describe('module PaperUI.controllers.things', function() {
...
-        });
-        it('should create item and link', function() {
-            spyOn(itemService, 'create').and.returnValue(deferred.promise).and.callThrough();
-            spyOn(itemService.create({}).$promise, 'then').and.returnValue(deferred.promise);
-            scope.newItemName = "N";
-            scope.createAndLink();
-            expect(itemService.create).toHaveBeenCalled();
-        });
-    });
+    // describe('tests for LinkChannelDialogController', function() {
```
[the remark this answers is not in the document]
Comment (htreu) [reply extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers/controller.things.spec.js]: ...or fixing the test actually.

[the remark this answers is not in the document]
Comment (htreu) [reply extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers/controller.things.spec.js]: Sadly the code now has `new Set` which is an ES6 feature and not supported by our PhantomJS which runs the tests. I have to investigate on how to update PhantomSJ or switch to another browser, like Chrome headless.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java (hunk 2 above)
Comment (htreu) [reply bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java]: This is just a copy of the original method javaDoc. We have to ask @sjka aboiut the missing text here.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java (hunk 3 above)
Comment (htreu) [reply bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java]: Yes, they do. The suggested ProfileTypeUID however will not depend on the given itemType when it comes to trigger channels.

**Comment by htreu** [comment]
Test added again, fell free to comment or merge :-)

**Comment by triller-telekom** [comment]
LGTM now.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemProfileFactory.java [hunk 5]
```diff
@@ -114,4 +128,13 @@ public ProfileTypeUID getSuggestedProfileTypeUID(Channel channel, @Nullable Stri
         return SUPPORTED_PROFILE_TYPE_UIDS;
     }
 
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.DYNAMIC)
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemProfileFactory.java]: mandatory+dynamic does not really bring much value. Why not merely do `@Reference` here?

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java (hunk 2 above)
Comment (kaikreuzer) [reply bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java]: "... does not have any advice"

**Pull request merged**